### PR TITLE
Improvements / bug fixes

### PR DESCRIPTION
**Pull request by wir3z**
App:  Removed times zone since it was causing an exception with the J…SON timestamps ending in "Z"
Thermostat: Added a dead band to push the cool or heat temperature set point to allow a new set point to be set.
Doorbell: Added device debug to differentiate multiple devices in Hubitat logs.

QUESTION:
Does saving lastEvent in local time pose issues when a timezone change occurs?  Typically it's advisable to save in UTC, and display in local to avoid this.

### CODE REVIEW

File: sdm-api-app.groovy [hunk 1]
```diff
@@ -435,6 +435,8 @@ def processCameraTraits(device, details) {
     def imgRes = details?.traits?.get('sdm.devices.traits.CameraImage')?.maxImageResolution
     imgRes?.width ? sendEvent(device, [name: 'imgWidth', value: imgRes.width]) : null
     imgRes?.height ? sendEvent(device, [name: 'imgHeight', value: imgRes.height]) : null   
+    def room = details?.parentRelations?.getAt(0)?.displayName
```
Comment (dkilgore90) [inline sdm-api-app.groovy:438]: I think this is a lingering merge conflict that ended up as a duplicate? covered in `def processTraits`, lines 367-368

File: sdm-api-app.groovy [hunk 2]
```diff
@@ -548,7 +550,7 @@ void deleteEventSubscription() {
 }
 
 def logToken() {
-    log.debug("Access Token: ${state.googleAccessToken}")
+    logDebug("Access Token: ${state.googleAccessToken}")
```
Comment (dkilgore90) [inline sdm-api-app.groovy:553]: intentionally left this one as `log.debug` as it is triggered only by the "Debug button", so we want to log it regardless of the toggle -- can bump it to `log.info` so it's clearer to always log?

File: sdm-api-doorbell.groovy [hunk 3]
```diff
@@ -77,12 +77,12 @@ def refresh() {
 }
 
 def processChime() {
-    logDebug('Doorbell/chime pressed')
+    logDebug('${device.label}: Doorbell/chime pressed')
```
Comment (dkilgore90) [inline sdm-api-doorbell.groovy:80]: as discussed in #32, doubling up on ${device.label} here (and all similar logDebug lines)

File: sdm-api-thermostat.groovy [hunk 4]
```diff
@@ -95,30 +95,46 @@ def off() {
...
-        parent.deviceSetTemperatureSetpoint(device, temp, device.currentValue('coolingSetpoint'))
+        parent.deviceSetTemperatureSetpoint(device, temp, tempMovement)
     } else {
         log.warn("Cannot setHeatingSetpoint in thermostatMode: ${mode}")
     }
 }
 
 def setHeatCoolSetpoint(heat, cool) {
+    def tempMovement = heat.toFloat() - cool.toFloat() + 1.5
+    if (tempMovement < 0) {
```
Comment (dkilgore90) [inline sdm-api-thermostat.groovy:133]: do we need this? check below is (tempMovement <= 0)

File: sdm-api-app.groovy (hunk 1 above)
Comment (wir3z) [reply sdm-api-app.groovy:438]: Agreed, my error.

File: sdm-api-app.groovy (hunk 2 above)
Comment (wir3z) [reply sdm-api-app.groovy:553]: Understood, just thought it was typo.

File: sdm-api-doorbell.groovy (hunk 3 above)
Comment (wir3z) [reply sdm-api-doorbell.groovy:80]: My mistake, it is covered in the debug function.

File: sdm-api-thermostat.groovy (hunk 4 above)
Comment (wir3z) [reply sdm-api-thermostat.groovy:133]: No need for this.  Cut/paste issue from the previous functions.

**Pull request closed without merging**